core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -400,15 +398,16 @@ private[kafka] class Processor(val id: Int,
 
   private val newConnections = new ConcurrentLinkedQueue[SocketChannel]()
   private val inflightResponses = mutable.Map[String, RequestChannel.Response]()
-  private val metricTags = Map("networkProcessor" -> id.toString).asJava
+  val metricTags = SortedMap("listener" -> listenerName.value, "networkProcessor" -> id.toString).asJava
```

Comment:
A LinkedHashMap is better because it makes it obvious that the order of declaration is important. With a SortedMap, adding a third key could change the order.